cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/selector/StorageSelector.java

```diff
@@ -0,0 +1,28 @@
+package com.linkedin.openhouse.cluster.storage.selector;
+
+import com.linkedin.openhouse.cluster.storage.Storage;
+
+/**
+ * The StorageSelector interface provides a way to select storage per table
+ *
+ * <p>Implementations of this interface can choose to return different storages for different db and
+ * tables For Example {@link DefaultStorageSelector} returns the cluster level storage for all
+ * tables
+ */
+public interface StorageSelector {
+  /**
+   * Select the storage for given db and table
+   *
```

Comment:
Can you make a note that this call should be idempotent? same db, table should always return same storage

Reply:
done